opm/output/eclipse/RestartIO.cpp

```diff
@@ -476,6 +557,52 @@ namespace {
         }
     }
 
+    void writeDynamicDataLGR(const int                                     sim_step,
+                             const EclipseGrid&                            grid,
+                             const EclipseState&                           es,
+                             const Schedule&                               schedule,
+                             const data::Wells&                            wellSol,
+                             const Opm::Action::State&                     action_state,
+                             const Opm::WellTestState&                     wtest_state,
+                             const Opm::SummaryState&                      sumState,
+                             const std::vector<int>&                       inteHD,
+                             EclIO::OutputStream::Restart&                 rstFile,
+                             const std::string&                            lgr_tag)
+    {
+        writeGroupLGR(sim_step, schedule.getUnits(), schedule, sumState, inteHD, rstFile, lgr_tag);
+
+        // Write network data if the network option is used and network defined
+        const auto& network = schedule[sim_step].network();
+        if (network.active())
+        {
+            writeNetwork(es, sim_step, schedule.getUnits(), schedule, sumState, inteHD, rstFile);
+        }
+
+        // Write well and MSW data only when applicable (i.e., when present)
+        if (const auto& wells = schedule.wellNames(sim_step);
+            ! wells.empty())
+        {
+            const auto haveMSW =
+            std::any_of(std::begin(wells), std::end(wells),
+            [&schedule, sim_step](const std::string& well)
+            {
+                auto lwell = schedule.getWell(well, sim_step);
```

Comment:
This copies a full `Well` object.  That's wasteful when you only read from the object.  Please rewrite this
as
```C++
const auto& lwell = schedule.getWell(well, sim_step);
```
instead.